customer-commons/src/test/java/no/unit/nva/customer/service/impl/DynamoDBCustomerServiceTest.java

```diff
@@ -0,0 +1,207 @@
+package no.unit.nva.customer.service.impl;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.document.Item;
+import com.amazonaws.services.dynamodbv2.document.Table;
+import com.amazonaws.services.dynamodbv2.local.embedded.DynamoDBEmbedded;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import no.unit.nva.customer.CustomerDynamoDBLocal;
+import no.unit.nva.customer.ObjectMapperConfig;
+import no.unit.nva.customer.exception.DynamoDBException;
+import no.unit.nva.customer.exception.InputException;
+import no.unit.nva.customer.exception.NotFoundException;
+import no.unit.nva.customer.model.Customer;
+import no.unit.nva.customer.service.CustomerService;
+import nva.commons.utils.Environment;
+import org.junit.Rule;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.migrationsupport.rules.EnableRuleMigrationSupport;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@EnableRuleMigrationSupport
+public class DynamoDBCustomerServiceTest {
+
+    public static final String NVA_CUSTOMERS_TEST = "nva_customers_test";
+    @Rule
+    public CustomerDynamoDBLocal db =  new CustomerDynamoDBLocal();
+
+    private ObjectMapper objectMapper = new ObjectMapperConfig().objectMapper;
+    private DynamoDBCustomerService service;
+    private Environment environment;
+    private AmazonDynamoDB client;
+
+    /**
+     * Set up environment.
+     */
+    @BeforeEach
+    public void setUp() {
+        client = DynamoDBEmbedded.create().amazonDynamoDB();
+        environment = mock(Environment.class);
+        service = new DynamoDBCustomerService(
+                objectMapper,
+                db.getTable()
+        );
+    }
+
+    @Test
+    public void testConstructor() {
+        when(environment.readEnv(DynamoDBCustomerService.TABLE_NAME)).thenReturn(NVA_CUSTOMERS_TEST);
+        CustomerService serviceWithTableNameFromEnv = new DynamoDBCustomerService(client, objectMapper, environment);;
+        assertNotNull(serviceWithTableNameFromEnv);
+    }
+
+    @Test
+    public void createNewCustomer() throws Exception {
+        Customer customer = getNewCustomer();
+        Customer createdCustomer = service.createCustomer(customer);
+
+        Assertions.assertNotNull(createdCustomer.getIdentifier());
+        Assertions.assertEquals(customer, createdCustomer);
+    }
+
+    @Test
+    public void updateExistingCustomer() throws Exception {
```

Comment:
This test should be broken to many tests. One for the equality, one for the modified date  and one for the created date. Although in previous cases it is OK to have multiple assertions in one test (e.g. when creatingCustomer we check the response body and the headers in one test), here it is less obvious what are the speciifcations and it should be broken to many tests